### PR TITLE
941: Warn about required merge after dependent PR integration

### PR DESCRIPTION
**Pull request by rwestberg**
When a dependent PR has been integrated, post an instructional message on how to merge these changes.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [SKARA-941](https://bugs.openjdk.java.net/browse/SKARA-941): Warn about required merge after dependent PR integration

### Reviewers
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to a8eec304d23cbf45c8ed41c46b7729532c563292

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/skara pull/1134/head:pull/1134` \
`$ git checkout pull/1134`

Update a local copy of the PR: \
`$ git checkout pull/1134` \
`$ git pull https://git.openjdk.java.net/skara pull/1134/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1134`

View PR using the GUI difftool: \
`$ git pr show -t 1134`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/skara/pull/1134.diff">https://git.openjdk.java.net/skara/pull/1134.diff</a>

</details>

### CODE REVIEW

**Comment by rwestberg** [comment]
The message looks like:

The dependent pull request has now been integrated, and the target branch of this pull request has been updated. This means that changes from the dependent pull request can start to show up as belonging to this pull request, which may be confusing for reviewers. To remedy this situation, simply merge the latest changes from the target branch into this pull request by running commands similar to these in the local repository for your personal fork:

```bash
git checkout source
git fetch file:///private/var/folders/d6/1lm60n2x37z35d9y72zl88k80000gn/T/RepositoryTests11763809428403690025/hosted.git master
git merge FETCH_HEAD
# if there are conflicts, follow the instructions given by git merge
git commit -m "Merge master"
git push
```

File: bots/notify/src/main/java/org/openjdk/skara/bots/notify/prbranch/PullRequestBranchNotifier.java [hunk 1]
```diff
@@ -76,8 +76,26 @@ public void onNewPullRequest(PullRequest pr, Path scratchPath) {
...
         if (pr.state() == Issue.State.CLOSED) {
-            PreIntegrations.retargetDependencies(pr);
+            var retargetedDependencies = PreIntegrations.retargetDependencies(pr);
             deleteBranch(pr);
+            for (var retargeted : retargetedDependencies) {
+                retargeted.addComment("""
+                    The dependent pull request has now been integrated, and the target branch of this pull request \
+                    has been updated. This means that changes from the dependent pull request can start to show up \
+                    as belonging to this pull request, which may be confusing for reviewers. To remedy this situation, \
+                    simply merge the latest changes from the target branch into this pull request by running commands \
```
Comment (erikj79) [inline bots/notify/src/main/java/org/openjdk/skara/bots/notify/prbranch/PullRequestBranchNotifier.java:86]:
```suggestion
                    simply merge the latest changes from the new target branch into this pull request by running commands \
```

**Review by erikj79 — APPROVED** [review]

**Comment by rwestberg** [comment]
@erikj79 Thanks for reviewing!

**Pull request closed without merging**